src/test/java/IntegrationTests/WAStorageClientDownloadIT.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+
+/**
+ *
+ * @author arroyc
+ */
+public class WAStorageClientDownloadIT extends IntegrationTest{
+    
+    private static final Logger LOGGER = Logger.getLogger(WAStorageClientDownloadIT.class.getName());
+    
+    /**
+     * Test of validateStorageAccount method, of class WAStorageClient.
+     */
+    @Test
+    public void testValidateStorageAccount() throws Exception {
+        System.out.println("validateStorageAccount");
+        StorageAccountInfo storageAccount = testEnv.sampleStorageAccount;
+        boolean expResult = true;
+        boolean result = WAStorageClient.validateStorageAccount(storageAccount);
+        assertEquals(expResult, result);
+    }
+
+    @Test(timeout=1200000)
+    public void testDownloadfromContainer() {
+        System.out.print("download");
+        CloudBlobContainer container;
+        try{
+            CloudStorageAccount account = new CloudStorageAccount(new StorageCredentialsAccountAndKey(testEnv.azureStorageAccountName, testEnv.azureStorageAccountKey1));
+            CloudBlobClient blobClient = account.createCloudBlobClient();
+            String containerName = "testdownload"+TestEnvironment.GenerateRandomString(15);
+            File directory = new File(containerName);
+            directory.mkdir();
+            container = blobClient.getContainerReference(containerName);
+            container.deleteIfExists();
+            container.createIfNotExists();
+            for(int i=0; i<testEnv.TOTAL_FILES;i++){
+                testEnv.downloadFileList.add(new File(directory, "download"+UUID.randomUUID().toString()+".txt"));
+                FileUtils.writeStringToFile(testEnv.downloadFileList.get(i), TestEnvironment.GenerateRandomString(15));
+                LOGGER.log(Level.INFO, testEnv.downloadFileList.get(i).getAbsolutePath());
+                CloudBlockBlob blob = container.getBlockBlobReference(testEnv.downloadFileList.get(i).getName());
+                blob.uploadText(TestEnvironment.GenerateRandomString(20), "UTF-8", AccessCondition.generateEmptyCondition(), null, null);
+            }                         
+            //try to download the same file     
+            Run mockRun = mock(Run.class);
+            Launcher mockLauncher = mock(Launcher.class);
+            WAStorageClient mockStorageClient = spy(WAStorageClient.class);
+            
+            FilePath workspace= new FilePath(mockLauncher.getChannel(), FilenameUtils.getFullPathNoEndSeparator(testEnv.downloadFileList.get(0).getAbsolutePath()));
+            int totalFiles = mockStorageClient.download(mockRun, mockLauncher, TaskListener.NULL, testEnv.sampleStorageAccount, "*.txt", ",archive.zip", "", false, workspace, containerName);
```

Comment:
why are you excluding archive.zip? I guess you should also have a test to make sure the exclude works